src/sagas/walletConnect.js

```diff
@@ -0,0 +1,424 @@
+    proposer: get(requestSession.peer, 'metadata.name', ''),
+    url: get(requestSession.peer, 'metadata.url', ''),
+    description: get(requestSession.peer, 'metadata.description', ''),
+  };
+
+  switch(params.request.method) {
+    case 'hathor_signMessage':
+      yield put({
+        type: 'SIGN_MESSAGE_REQUEST',
+        payload: {
+          ...data,
+          requestId: payload.id,
+          topic: payload.topic,
+          message: get(params, 'request.params.message'),
+        }
+      });
+    break;
+    default:
+      yield call(() => web3wallet.respondSessionRequest({
+        topic: payload.topic,
+        response: {
+          id: payload.id,
+          jsonrpc: '2.0',
+          error: {
+            code: USER_REJECTED_METHOD,
+            message: 'Rejected by the user',
+          },
+        },
+      }));
+    break;
+  }
+}
+
+export function* onSignMessageRequest(action) {
+  const data = action.payload;
+  const { web3wallet } = yield select((state) => state.walletConnect);
+
+  const onAcceptAction = { type: 'WALLET_CONNECT_ACCEPT' };
+  const onRejectAction = { type: 'WALLET_CONNECT_REJECT' };
+
+  yield put(setWalletConnectModal({
+    show: true,
+    type: WalletConnectModalTypes.SIGN_MESSAGE,
+    data,
+    onAcceptAction,
+    onRejectAction,
+  }));
+
+  const { accept } = yield race({
+    accept: take(onAcceptAction.type),
+    reject: take(onRejectAction.type),
+  });
+
+  try {
+    if (!accept) {
+      yield call(() => web3wallet.respondSessionRequest({
+        topic: payload.topic,
+        response: {
+          id: payload.id,
+          jsonrpc: '2.0',
+          error: {
+            code: USER_REJECTED,
+            message: 'Rejected by the user',
+          },
+        },
+      }));
+      return;
+    }
+
+    const wallet = yield select((state) => state.wallet);
+
+    if (!wallet.isReady()) {
+      console.error('Got a session request but wallet is not ready, ignoring..');
+      return;
+    }
+
+    const signedMessage = yield call(() => wallet.signArbitraryMessage(data.message));
+
+    const response = {
+      id: data.requestId,
+      jsonrpc: '2.0',
+      result: signedMessage,
+    };
+
+    yield call(() => web3wallet.respondSessionRequest({
+      topic: data.topic,
+      response,
+    }));
+  } catch(error) {
+    console.log('Captured error on signMessage: ', error);
+    yield put(onExceptionCaptured(error));
+  }
+}
+
+export function* onWalletReset() {
+  const { web3wallet } = yield select((state) => state.walletConnect);
+  if (!web3wallet) {
+    // Do nothing, wallet connect might not have been initialized yet
+    return;
+  }
+
+  yield call(clearSessions);
+}
+
+export function* onSessionProposal(action) {
+  const { id, params } = action.payload;
+  const { web3wallet } = yield select((state) => state.walletConnect);
+
+  const wallet = yield select((state) => state.wallet);
+  const addresses = wallet.newAddresses;
```

Comment:
You don't have this in the old facade. If you need only the base58 string you can use `wallet.getAddressAtIndex(0)`